src/client/api.ts

```diff
@@ -32,7 +33,13 @@ export interface IExtensionApi {
      * @memberof IExtensionApi
      */
     ready: Promise<void>;
+    /**
+     * Do not use this to monitor execution state of cells of Native Notebooks (use VS Code API).
+     */
     readonly onKernelStateChange: Event<KernelStateEventArgs>;
```

Comment:
@DavidKutu You'll need to update Gather extension to use this new VS Code API to get it work with the Jupyter extension. I.e. use `notebook.onDidChangeCellExecutionState` event

Reply:
This is needed for the Interactive window

Reply:
All I'm saying here is Don't use this for Native Notebooks, & if using this for anythign else then continue doing so (I didn't want to mention what this is for or where it can be used)